src/ui/RealmBrowser/Content/CreateObjectDialog/types/ObjectIdControl.tsx

```diff
@@ -0,0 +1,105 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import React from 'react';
+import { ObjectId } from 'bson';
+import { Button, Input, InputGroup, InputGroupAddon } from 'reactstrap';
+
+import { IBaseControlProps } from './TypeControl';
+import { parseObjectId } from '../../../parsers';
+
+interface IObjectIdControlState {
+  internalValue: string | null;
+}
+
+export class ObjectIdControl extends React.PureComponent<
+  IBaseControlProps,
+  IObjectIdControlState
+> {
+  constructor(props: IBaseControlProps) {
+    super(props);
+
+    this.state = {
+      internalValue: props.value,
+    };
```

Comment:
Historically we've [initialized the state when defining it as a member](https://github.com/realm/realm-studio/blob/master/src/ui/RealmBrowser/Content/CreateObjectDialog/types/ObjectControl/index.tsx#L51-L53) - I do however see the benefit that with this approach you don't need to write the type twice.

Reply:
Since this is simply used to grab the initial default value of the property, there is no practical issue here - but we could run into a weird error if we used this component in a context where the property would be updated and we wanted to update our `internalValue` accordingly.